src/main/fc/fc_core.c

```diff
@@ -389,13 +402,61 @@ static bool canUpdateVTX(void)
 }
 #endif
 
+// determine if the R/P/Y stick deflection exceeds the specified limit - integer math is good enough here.
+uint8_t areSticksActive(uint8_t stickPercentLimit) {
+    uint8_t deadband;
+    uint8_t stickPercent;
+    bool result = false;
+
+    for (int axis = FD_ROLL; axis <= FD_YAW; axis ++) {
+        if (axis == FD_YAW) {
+            deadband = rcControlsConfig()->yaw_deadband;
+        } else {
+            deadband = rcControlsConfig()->deadband;
+        }
+        if ((rcData[axis] >= PWM_RANGE_MAX) || (rcData[axis] <= PWM_RANGE_MIN)) {
+            stickPercent = 100;
+        } else {
+            if (rcData[axis] > (rxConfig()->midrc + deadband)) {
+                stickPercent = ((rcData[axis] - rxConfig()->midrc - deadband) * 100) / (PWM_RANGE_MAX - rxConfig()->midrc - deadband);
+            } else if (rcData[axis] < (rxConfig()->midrc - deadband)) {
+                stickPercent = ((rxConfig()->midrc - deadband - rcData[axis]) * 100) / (rxConfig()->midrc - deadband - PWM_RANGE_MIN);
+            } else {
+                stickPercent = 0;
+            }
+        }
+        if (stickPercent >= stickPercentLimit) {
+            result = true;
+        }
+    }
+    return result;
+}
+
+
+// calculate the throttle percent based on stick position in the range from min_check to PWM_RANGE_MAX - integer math is good enough here.
+// TODO - make this work for 3D mode as well
+uint8_t calculateThrottlePercent(void) {
+    uint8_t result;
+
+    if (rcData[THROTTLE] <= rxConfig()->mincheck) {
+       result = 0;
+    } else if (rcData[THROTTLE] >= PWM_RANGE_MAX) {
+       result = 100;
+    } else {
+       result = ((rcData[THROTTLE] - rxConfig()->mincheck) * 100) / (PWM_RANGE_MAX - rxConfig()->mincheck);
+    }
+    return result;
```

Comment:
```
const uint ret = constrain(100*(rcData[THROTTLE] - rxConfig()->mincheck) / (PWM_RANGE_MAX - rxConfig()->mincheck), 0, 100);
```

Reply:
This is a good simplification.  Originally did it literally because I knew I had to add 3D support later wanted it easy to understand.  Also because of upcoming 3D support I left the `ret` variable in place rather than just returning the calculation.  It will get more complex having to compare the 3D throttle deadbands, etc.

```
uint8_t calculateThrottlePercent(void)
{
    uint8_t ret;

    ret = constrain(((rcData[THROTTLE] - rxConfig()->mincheck) * 100) / (PWM_RANGE_MAX - rxConfig()->mincheck), 0, 100);
    return ret;
}
```